micro-rdk/src/common/ota.rs

```diff
@@ -159,11 +164,41 @@ impl OtaService {
                 )))
             }
         };
+
+        #[cfg(not(feature = "esp32"))]
+        let (max_size, address, label) = (1024 * 1024 * 4, 0, "debug".to_string());
+        #[cfg(feature = "esp32")]
+        let (max_size, address, label) = {
+            log::debug!("getting handle to next OTA update partition");
+            let ptr: *const esp_partition_t =
+                unsafe { esp_ota_get_next_update_partition(std::ptr::null()) };
+            if ptr.is_null() {
+                return Err(OtaError::Other(
+                    "failed to obtain a handle to the next OTA update partition".to_string(),
+                ));
+            }
+            let size = unsafe { (*ptr).size };
+            let address = unsafe { (*ptr).address };
+            let label = {
+                // char[17] zero-terminated ascii string
+                let label_ptr = unsafe { (*ptr).address as *const std::ffi::c_char };
+                let cstr = unsafe { std::ffi::CStr::from_ptr(label_ptr) };
+                cstr.to_str()
+                    .inspect_err(|e| log::error!("failed to get label information: {}", e))
+                    .unwrap_or_default()
+                    .to_owned()
+            };
+            (size, address, label)
```

Comment:
@acmorrow added some extra info while I figured would be good to surface about the target partition.

need to verify this works consistently on hardware before I merge it.

Reply:
my attempts at getting the `label` led to invalid accesses. I'm sure there are ways to do it properly but I'm removing it for now as it's not essential to the PR